app/src/main/java/org/mozilla/fenix/FenixApplication.kt

```diff
@@ -402,7 +404,27 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
             // experiment on features close to startup.
             // But we need viaduct (the RustHttp client) to be ready before we do.
             components.analytics.experiments.initialize()
+
+            setupMessaging()
+        }
+    }
+
+    private fun setupMessaging() {
+        if (!FeatureFlags.messagingFeature) {
+            return
+        }
+
+        fun restore() {
+            components.appStore.dispatch(AppAction.MessagingAction.Restore)
         }
+
+        components.analytics.experiments.register(object : NimbusInterface.Observer {
+            override fun onUpdatesApplied(updated: List<EnrolledExperiment>) {
+                restore()
+            }
+        })
```

Comment:
Is it  Ok if we added as part of the existing [onUpdatesApplied]( https://github.com/mozilla-mobile/fenix/pull/25266/files#diff-9e2b3999e8d0e181af6decd7d25949eef361031a719a578869967a5339b31737R772)?

Reply:
I combined the two observers into one.